File: clj/001.clj
;; Project Euler - Problem 1
;; Find the sum of all the multiples of 3 or 5 below 1000.

(letfn [(multiple-of? [a b]
          (= 0 (rem a b)))]
  (apply + (filter #(or (multiple-of? % 3) (multiple-of? % 5)) (range 1000))))

File: pl/001.pl
#!/usr/bin/perl

# Project Euler - Problem 1
# Find the sum of all the multiples of 3 or 5 below 1000.

use 5.10.0;

my $sum = 0;
$sum += $_ % 3 && $_ % 5 ? 0 : $_
    for 3 .. 999;
say $sum;
